src/arrayMethodJoin.js

```diff
@@ -5,7 +5,26 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (separator === undefined) {
+      return this.join2(',');
+    }
+
+    let joinedString = '';
+    const arrayLength = this.length;
+
+    for (let e = 0; e < arrayLength; e++) {
```

Comment:
In most cases the convention is to name the iterator variable **i**

Reply:
I renamed variable to _e_.